src/base/LruMap.h

```diff
@@ -135,11 +153,29 @@ LruMap<EntryValue, EntryCost>::get(const char *key)
     return NULL;
 }
 
-template <class EntryValue, size_t EntryCost>
+template <class EntryValue, size_t MemoryUsedByEV(const EntryValue *)>
 bool
-LruMap<EntryValue, EntryCost>::add(const char *key, EntryValue *t)
+LruMap<EntryValue, MemoryUsedByEV>::add(const char *key, EntryValue *t)
 {
-    if (ttl == 0)
+    // map the retention concept into a TTL value
+
+    // do not store / retain any entries
+    if (retention == 0)
+        return false;
+
+    // negative retention means store 'forever'
+    if (retention < 0)
+        return add(key, t, 0xFFFFFFFF);
+
+    // positive retention are TTL
+    return add(key, t, retention);
+}
+
+template <class EntryValue, size_t MemoryUsedByEV(const EntryValue *)>
+bool
+LruMap<EntryValue, MemoryUsedByEV>::add(const char *key, EntryValue *t, int ttl)
+{
+    if (retention == 0)
```

Comment:
Please either add a comment explaining why we store expired (i.e., `ttl <= 0`) values or do not store them.